src/generate-oas.ts

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import path from "path";
+import fs from "fs-extra";
+import { generateFromOas, download } from "@commerce-apps/raml-toolkit";
+import Handlebars from "handlebars";
+
+type ApiSpecDetail = {
+  filepath: string;
+  filename: string;
+  name: string;
+  apiName: string;
+  directoryName: string;
+};
+
+const API_DIRECTORY = path.resolve(`${__dirname}/../apis`);
+const STATIC_DIRECTORY = path.join(__dirname, "./static");
+const TARGET_DIRECTORY = path.resolve(`${__dirname}/../renderedTemplates`);
+const TEMPLATE_DIRECTORY = path.resolve(`${__dirname}/../templatesOas`);
+const INDEX_TEMPLATE_LOCATION = path.resolve(
+  `${__dirname}/../templates/index.ts.hbs`
+);
+const VERSION_TEMPLATE_LOCATION = path.resolve(
+  `${__dirname}/../templates/version.ts.hbs`
+);
+
+function kebabToCamelCase(str: string): string {
+  return str.replace(/-([a-z])/g, (match, letter: string) =>
+    letter.toUpperCase()
+  );
+}
+
+/**
+ * Helper function. Also contains explicit workaround for some API names.
+ */
+export function resolveApiName(name: string): string {
+  if (name === "Shopper orders OAS") {
+    return "ShopperOrders";
+  }
+  if (name === "CDN API - Process APIs OAS") {
+    return "CDNZones";
+  }
+  if (name === "SLAS Admin-UAP OAS") {
+    return "SlasAdmin";
+  }
+  if (name === "Shopper Seo OAS") {
+    return "ShopperSEO";
+  }
+  if (name === "Shopper Context OAS") {
+    return "ShopperContexts";
+  }
+  if (name === "Catalogs OAS") {
+    return "CatalogsAPI";
```

Comment:
Do you have any strong preferences between `CatalogsAPI` vs `CatalogsV1`? Since we're already using the `V2` convention for shopper baskets

Reply:
No preference. I'll change this to `CatalogsV1` to match the convention with baskets